core/envelope-sim/src/main/kotlin/fr/sncf/osrd/envelope_sim/etcs/ETCSBrakingCurves.kt

```diff
@@ -344,21 +341,35 @@ private fun computeIndicationBrakingCurveFromRef(
 
 /**
  * Keep the part of the full braking curve which is located underneath the overlay and intersects
- * with it or with begin position. If the part has no intersection, return null. Ex: the part only
- * has negative positions, and the overlay starts at 0.0.
+ * with it or with begin position. The curve should always intersect, throw otherwise.
  */
 private fun keepBrakingCurveUnderOverlay(
-    fullBrakingCurve: EnvelopePart,
+    fullBrakingCurve: Envelope,
     overlay: Envelope,
     beginPos: Double
-): EnvelopePart? {
-    if (fullBrakingCurve.endPos <= beginPos) {
-        return null
-    }
-    val positions = fullBrakingCurve.clonePositions()
-    val speeds = fullBrakingCurve.cloneSpeeds()
-    val timeDeltas = fullBrakingCurve.cloneTimes()
-    val nbPoints = fullBrakingCurve.pointCount()
+): EnvelopePart {
```

Comment:
Especially for EoA and LoA, I would check before calling that there is no need to add a curve if no intersection.

Opinionated (so OK to dismiss): From the `keepBrakingCurveUnderOverlay` fn itself, it's a legitimate use IMO to have no intersection.
I would let the responsibility to the caller to check if it is an expected case.
```suggestion
): EnvelopePart? {
```

Reply:
To check whether or not there is an intersection between 2 envelopes is not something so easily done, especially when they're not both continuous and monotonous. However, yeah, as we talked about it, the current flow rn is not to throw exceptions, so I would rather return null here, and log a warning in the above method if it is null.